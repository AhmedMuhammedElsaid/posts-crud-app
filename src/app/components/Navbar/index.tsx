import Link from "next/link";
import classes from "./styles.module.css";

const Navbar = () => {
  return (
    <nav className={classes.Nav}>
      <Link href="/" passHref>
        <p className={classes.NavLink}>Posts</p>
      </Link>
      <Link href="/post/create" passHref>
        <p className={classes.NavLink}>Create Post</p>
      </Link>
    </nav>
  );
};

export default Navbar;
