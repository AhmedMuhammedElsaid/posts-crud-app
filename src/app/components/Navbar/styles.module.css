.Nav {
  background-color: #333;
  color: #fff;
  display: flex;
  justify-content: space-around;
  padding: 15px 0;
  height: 80px;
  align-items: center;
}

.NavLink {
  color: #fff;
  text-decoration: none;
  transition: color 0.3s ease;
  font-weight: 600;
}
.NavLink:hover {
  color: #0070f3;
}
