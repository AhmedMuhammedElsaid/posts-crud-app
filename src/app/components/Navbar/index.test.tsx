import React from "react";
import { render, screen } from "@testing-library/react";
import { useRouter } from "next/router";
import Navbar from "./index";

jest.mock("next/router", () => ({
  useRouter: () => ({
    route: "/",
    pathname: "/",
    query: "",
    asPath: "/",
  }),
}));

test("renders navbar", () => {
  const { asFragment } = render(<Navbar />);
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByText("Posts")).toBeInTheDocument();
  expect(screen.getByText("Create Post")).toBeInTheDocument();
});
