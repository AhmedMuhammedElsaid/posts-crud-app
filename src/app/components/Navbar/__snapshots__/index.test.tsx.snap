// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar 1`] = `
<DocumentFragment>
  <nav
    class="Nav"
  >
    <a
      href="/"
    >
      <p
        class="NavLink"
      >
        Posts
      </p>
    </a>
    <a
      href="/post/create"
    >
      <p
        class="NavLink"
      >
        Create Post
      </p>
    </a>
  </nav>
</DocumentFragment>
`;
